### PR TITLE
Add suite with alcotest

### PR DESCRIPTION
**Pull request by gaborigloi**
And test Valid_ref_list using alcotest.

Signed-off-by: Gabor Igloi <gabor.igloi@citrix.com>

### CODE REVIEW

File: ocaml/xapi/jbuild [hunk 1]
```diff
@@ -112,4 +113,11 @@ let () = Printf.ksprintf Jbuild_plugin.V1.send {|
   (action (run ${<} -runner sequential -verbose true))
  )
 )
+
+(alias
+ ((name runalcotest)
```
Comment (edwintorok) [inline ocaml/xapi/jbuild]: I think you can make both of these run as part of runtest if you put alcotest in another folder

Comment (gaborigloi) [reply ocaml/xapi/jbuild]: How can I do that?

Comment (edwintorok) [reply ocaml/xapi/jbuild]: See https://discuss.ocaml.org/t/jbuilder-support-for-multiple-test-executables/825 you can in fact have two `runtest` in same jbuild file.

Comment (gaborigloi) [reply ocaml/xapi/jbuild]: Thanks, that's great, I'll use two!

**Comment by lindig** [comment]
xs-opam has been released. Can we merge this?

**Comment by mseri** [comment]
Let's rerun travis first, once we have a build with the next `xs-opam` uploaded to the xs-buildenv

**Comment by mseri** [comment]
Same issue as `message-switch`... There is finally a plan upstream to fix the `ppx_bisect`/coverage situation. Until that is complete you only need to install by hand the test dependencies in the `.coverage` script. It's a matter of adding `alcotest` to the `opam install -y oUnit ...` line

**Comment by gaborigloi** [comment]
How about using `opam install --build-test`?

**Comment by mseri** [comment]
Try. It did not work when I tried on message-switch.

**Comment by gaborigloi** [comment]
Oh, that doesn't work because that tries to install the test deps of the dependencies transitively

**Pull request merged**